HashRegistrar.sol

```diff
@@ -0,0 +1,374 @@
+just holding it expecting that domains names will become more valuable in 
+the future. Let's assume that all name buyers have the intention of acquiring 
+a name and make it more valuable over time, either by working on it as a 
+business and adding value to the "brand", or by working to increase the 
+chances of finding a better suited buyer, all have variable success on these 
+endeavours. It's natural to assume that the value of new names being acquired 
+should keep in line with the market values expectation of how profitable they 
+are to sell (either as a brand, a business or a placeholder domain) in the future.
+
+The solution here lies to require a deposit to own a name and periodically 
+require the owner to update their deposit to what is assumed to be the market 
+rate of the name of their current name. If names in general have increased in 
+value but the owner is among the bottom half that hasn't done anything to increase 
+the value of their own names, then all possible profit will be negated by the extra 
+deposit. If names have decreased in value the owners can request to withdraw part 
+of the deposit and the cost of the name will be the cost of opportunity of possibly 
+having invested in something that would have a better return. At any point name 
+holders can release and get the full deposit back.
+
+The contract is called hash registrar because it deals with ownership of hashes of 
+things, never the things themselves, to increase privacy and extensibility.
+
+*/
+
+contract Deed {
+    /* 
+    The Deed is a contract intended simply to hold ether
+    It can be controlled only by the registrar and can only send ether back to the owner.
+    */
+    Registrar public registrar;
+    address constant burn = 0xdead;
+    uint public creationDate;
+    address public owner;
+    event OwnerChanged(address newOwner);
+    event DeedClosed();
+    bool active;
+
+    modifier onlyRegistrar {
+        if (msg.sender != address(registrar)) throw;
+        _
+    }
+
+    modifier onlyActive {
+        if (!active) throw;
+        _
+    }
+    
+    function Deed() {
+        registrar = Registrar(msg.sender);
+        creationDate = now;
+        active = true;
+    }
+        
+    function setOwner(address newOwner) {
+        if ((owner > 0 && msg.sender != owner )
+            || (owner == 0 && msg.sender != address(registrar))) throw;
+        owner = newOwner;
+        OwnerChanged(newOwner);
+    }
+    
+    function setBalance(uint newValue) onlyRegistrar onlyActive {
+        // Check if it has enough balance to set the value
+        if (this.balance < newValue) throw;
+        // Send the difference to the owner
+        if (!owner.send(this.balance - newValue)) throw;
+    }
+
+    function closeDeed(uint refundRatio) onlyRegistrar onlyActive {
+        active = false;            
+        burn.send(((1000 - refundRatio) * this.balance)/1000);
```

Comment:
Do we need to worry about stack depth attack here? I can see a situation where a registrar and owner wants `closeDeed` to fail so that they can call `destroyDeed` to get `this.balance` thus they can avoid the burning.